chasm/lib/worker/tasks.go

```diff
@@ -0,0 +1,121 @@
+// Tasks that are scheduled for Workers and the corresponding executors.
+package worker
+
+import (
+	"time"
+
+	"go.temporal.io/server/chasm"
+	workerstatepb "go.temporal.io/server/chasm/lib/worker/gen/workerpb/v1"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+)
+
+// LeaseExpiryTaskExecutor handles lease expiry events.
+type LeaseExpiryTaskExecutor struct {
+	logger log.Logger
+	config *Config
+}
+
+func NewLeaseExpiryTaskExecutor(logger log.Logger, config *Config) *LeaseExpiryTaskExecutor {
+	return &LeaseExpiryTaskExecutor{
+		logger: logger,
+		config: config,
+	}
+}
+
+// Execute is called when a lease expiry timer fires.
+func (e *LeaseExpiryTaskExecutor) Execute(
+	ctx chasm.MutableContext,
+	worker *Worker,
+	attrs chasm.TaskAttributes,
+	task *workerstatepb.LeaseExpiryTask,
+) error {
+	// Validate that this lease expiry is still relevant.
+	if !e.isLeaseExpiryTaskValid(worker, attrs) {
```

Comment:
Don't need to validate a pure task, validation should happen under the same lock.